modules/dependencies.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.dependencies;
+
+  packages = {
+    ctags = "ctags";
+    curl = "curl";
+    direnv = "direnv";
+    distant = "distant";
+    fish = "fish";
+    gcc = "gcc";
+    gh = "gh";
+    git = "git";
+    go = "go";
+    lean = "lean4";
+    ledger = "ledger";
+    manix = "manix";
+    nodejs = "nodejs";
+    ripgrep = "ripgrep";
+    sd = "sd";
+    sed = "gnused";
+    texpresso = "texpresso";
+    tinymist = "tinymist";
+    treesitter = "tree-sitter";
+    typst = "typst";
+    ueberzug = "ueberzugpp";
+    websocat = "websocat";
+    which = "which";
+    yazi = "yazi";
+  };
+
+  mkDependencyOption = name: default: {
+    enable = lib.mkEnableOption "Add ${name} to dependencies.";
+
+    package = lib.mkPackageOption pkgs name { inherit default; };
```

Comment:
I did like the idea of being able to specify an `example` alongside the `default`. Happy to do it in a follow up PR if preferred.

If you just like not needing `.default =` for most cases, we could check the second param's type:

```nix
mkDependencyOption =
  name: arg:
  let
    # optional let-binding; could probably fit inline and still be readable.
    args = if builtins.isAttrs arg then arg else { default = arg; };
  in
```